cli/tests/generations.bats

```diff
@@ -0,0 +1,75 @@
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test rust impl of `flox pull`
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash
+# bats file_tags=push
+
+# ---------------------------------------------------------------------------- #
+
+# Helpers for project based tests.
+
+project_setup() {
+  export PROJECT_DIR="${BATS_TEST_TMPDIR?}/project-push-${BATS_TEST_NUMBER?}"
+  rm -rf "$PROJECT_DIR"
+  mkdir -p "$PROJECT_DIR"
+  pushd "$PROJECT_DIR" >/dev/null || return
+  export _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/empty.yaml"
+}
+
+project_teardown() {
+  popd >/dev/null || return
+  rm -rf "${PROJECT_DIR?}"
+  unset PROJECT_DIR
+}
+
+# ---------------------------------------------------------------------------- #
+
+setup() {
+  common_test_setup
+  setup_isolated_flox
+  project_setup
+  floxhub_setup "owner"
+}
+teardown() {
+  project_teardown
+  common_test_teardown
+}
+
+# ---------------------------------------------------------------------------- #
+
+@test "commands are displayed for generations history" {
+  mkdir -p "machine_a"
+  mkdir -p "machine_b"
+
+  "$FLOX_BIN" init --name "test"
+  "$FLOX_BIN" push --owner owner
+
+  # Make a few modifications
+  # 1. an edit
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+EOF
+  )"
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f -
+
+  # 2. an install
+  _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/resolve/hello.yaml" \
+    "$FLOX_BIN" install hello
+
+  # 3. switch generation, but set argv[0] to foo
+  (exec -a foo "$FLOX_BIN" generations switch 1)
+
+  run "$FLOX_BIN" generations history
+  # I'm not completely sure it's desirable, but this documents the current behavior:
+  # - If argv[0] is set to something other than flox, that's recorded
+  # - If argv[0] is a path, that's recorded
```

Comment:
We could just print `flox`

Reply:
Just `flox` seems fine. I'm not sure that we (or rather other users) need to know how they got there.